tests/components/forked_daapd/test_media_player.py

```diff
@@ -0,0 +1,726 @@
+    """Test async play media with TTS timeout."""
+    mock_api_object.add_to_queue.side_effect = None
+    with patch("homeassistant.components.forked_daapd.media_player.TTS_TIMEOUT", 0):
+        initial_state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+        await _service_call(
+            hass,
+            TEST_MASTER_ENTITY_NAME,
+            SERVICE_PLAY_MEDIA,
+            {
+                ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_MUSIC,
+                ATTR_MEDIA_CONTENT_ID: "somefile.mp3",
+            },
+        )
+        state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+        assert state.state == initial_state.state
+        assert state.last_updated > initial_state.last_updated
+
+
+async def test_use_pipe_control_with_no_api(hass, mock_api_object):
+    """Test using pipe control with no api set."""
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_SELECT_SOURCE,
+        {ATTR_INPUT_SOURCE: "librespot-java (pipe)"},
+    )
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_PLAY)
+    assert mock_api_object.start_playback.call_count == 0
+
+
+async def test_clear_source(hass, mock_api_object):
+    """Test changing source to clear."""
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_SELECT_SOURCE,
+        {ATTR_INPUT_SOURCE: SOURCE_NAME_CLEAR},
+    )
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.attributes[ATTR_INPUT_SOURCE] == SOURCE_NAME_DEFAULT
+
+
+@pytest.fixture(name="pipe_control_api_object")
+async def pipe_control_api_object_fixture(
+    hass, config_entry, get_request_return_values, mock_api_object
+):
+    """Fixture for mock librespot_java api."""
+    with patch(
+        "homeassistant.components.forked_daapd.media_player.LibrespotJavaAPI",
+        autospec=True,
+    ) as pipe_control_api:
+        hass.config_entries.async_update_entry(config_entry, options=OPTIONS_DATA)
+        await hass.async_block_till_done()
+    get_request_return_values["player"] = SAMPLE_PLAYER_PLAYING
+    updater_update = mock_api_object.start_websocket_handler.call_args[0][2]
+    await updater_update(["player"])
+    await hass.async_block_till_done()
+
+    async def pause_side_effect():
+        await updater_update(["player"])
+
+    pipe_control_api.return_value.player_pause.side_effect = pause_side_effect
+
+    await updater_update(["database"])  # load in sources
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_SELECT_SOURCE,
+        {ATTR_INPUT_SOURCE: "librespot-java (pipe)"},
+    )
+
+    return pipe_control_api.return_value
+
+
+async def test_librespot_java_stuff(hass, pipe_control_api_object):
+    """Test options update and librespot-java stuff."""
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.attributes[ATTR_INPUT_SOURCE] == "librespot-java (pipe)"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)